internal/mode/static/handler.go

```diff
@@ -92,3 +118,81 @@ func (h *eventHandlerImpl) updateNginx(ctx context.Context, conf dataplane.Confi
 
 	return nil
 }
+
+// updateControlPlane updates the control plane configuration with the given user spec.
+// If any fields are not set within the user spec, the default configuration values are used.
+func (h *eventHandlerImpl) updateControlPlane(ctx context.Context, cfg *nkgAPI.NginxGateway) {
+	// build up default configuration
+	defaultLogLevel := nkgAPI.ControllerLogLevelInfo
+	controlConfig := nkgAPI.NginxGatewaySpec{
+		Logging: &nkgAPI.Logging{
+			Level: &defaultLogLevel,
+		},
+	}
+
+	updateControlPlane := func() error {
+		// by marshaling the user config and then unmarshaling on top of the default config,
+		// we ensure that any unset user values are set with the default values
+		if cfg != nil {
+			cfgBytes, err := json.Marshal(cfg.Spec)
+			if err != nil {
+				return fmt.Errorf("error marshaling control config: %w", err)
+			}
+
+			if err := json.Unmarshal(cfgBytes, &controlConfig); err != nil {
+				return fmt.Errorf("error unmarshaling control config: %w", err)
+			}
+		} else {
+			msg := "NginxGateway configuration was deleted; using defaults"
+			h.cfg.logger.Error(nil, msg)
+			h.cfg.eventRecorder.Event(
+				&nkgAPI.NginxGateway{
+					ObjectMeta: metav1.ObjectMeta{
+						Namespace: h.cfg.controlConfigNSName.Namespace,
+						Name:      h.cfg.controlConfigNSName.Name,
+					},
+				},
+				apiv1.EventTypeWarning,
+				"ResourceDeleted",
+				msg,
+			)
+		}
+
+		// set the log level
+		level, err := zapcore.ParseLevel(string(*controlConfig.Logging.Level))
+		if err != nil {
+			return fmt.Errorf("error parsing log level string: %w", err)
+		}
+		h.cfg.atomicLevel.SetLevel(level)
+
+		return nil
+	}
+
+	var cond []conditions.Condition
+	if err := updateControlPlane(); err != nil {
+		msg := "Failed to update control plane configuration"
+		h.cfg.logger.Error(err, msg)
+		h.cfg.eventRecorder.Eventf(
```

Comment:
I like having an Event for resources that directly apply to us and that we manage. Otherwise, something might break and a user has no idea what happened unless they look at logs (and events are much easier at first glance).

Reply:
👍 

(1)
could you also add an info log message updateControlPlane so that we record that we reconfigured control plane?

(2)
additionally, would it make sense to change "FailedUpdate" to "UpdateFailed", because the other reason is "ResourceDeleted", for consistency?

(3)
If we log events, I think it makes sense to document that somewhere